alphagov/cf-terraform
# [#102465348] Create CF instance on GCE

We are currently able to deploy microBOSH to GCE thanks to [#100412960]. This enables us to build a full cloud foundry instance in this story.
# What
- Added fixed external IP for HA proxy
- Extended provision script to deploy cloud foundry
- Used cloud foundry manifest from:
  
  https://github.com/alphagov/cf-terraform/blob/feature/bosh-nats-gce/gce/cf210-manifest.yml.erb
  It is for release 210, Google cloud CPI
- Rendered cloud foundry manifest using ERB locally before uploading to BOSH
- Added actions to Makefile
# How to review
1. Create VMs, MicroBOSH and cloud foundry via:
   
   ```
   make gce DEPLOY_ENV=<environment>
   ```
   
   This should take about 1 hour.
2. After it has finished, you should be able to target the CF api using the static IP address that was assigned to the haproxy machine.
   
   ```
   $ cf api --skip-ssl-validation https://api.130.211.68.68.xip.io
   $ cf login
   API endpoint: https://api.130.211.65.123.xip.io
   Email> admin
   Password> c1ouc0w
   ```
3. Create an organisation and space within CF so that you can deploy a test app.
   
   ```
   $ cf create-org me
   $ cf target -o me
   $ cf create-space development
   $ cf target -o me -s development
   ```
4. Checkout a sample app to test the system with.
   
   ```
   $ git clone https://github.com/cloudfoundry-samples/spring-music.git
   $ cd spring-music
   $ ./gradlew assemble
   $ cf push
   ```

After a couple of minutes the app should be available for use. `cf apps` will show you the URL for your app.
# Who can review

Anyone but @mtekel and @saliceti
